### PR TITLE
CORE: GET /api/articles (topic query)

### PR DESCRIPTION
**Pull request by Mpanasetckiy**
fix(app.test.js): add tests for GET requests to `/api/articles` endpoit with different query parameters to ensure correct filtering and response

fix(articles.models.js): modify fetchArticles function to include topic query parameter in SQL query to filter articles by topic

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -107,6 +107,36 @@ describe("App endpoints", () => {
...
+        return request(app)
+          .get("/api/articles?topic=mitch")
+          .expect(200)
+          .then(({ body: { articles } }) => {
+            expect(articles).toHaveLength(12);
+            articles.forEach(({ topic }) => {
+              expect(topic).toBe("mitch");
+            });
+          });
+      });
```
Comment (nc-niamh-smith) [inline __tests__/app.test.js:121]: ^^ this is a great test - it's clear and clean and expects exactly what is suggested.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -107,6 +107,36 @@ describe("App endpoints", () => {
...
+
+      test("200 - GET: Responds with an empty array when no corresponding articles found", () => {
+        return request(app)
+          .get("/api/articles?topic=blahblahtopic")
+          .expect(200)
+          .then(({ body: { articles } }) => {
+            expect(articles).toHaveLength(0);
+          });
+      });
     });
```
Comment (nc-niamh-smith) [inline __tests__/app.test.js:140]: nice! What should happen if a topic exists but there are no associated articles (e.g. 'paper'!) ?

**Review by nc-niamh-smith — COMMENTED** [review]
Lovely work here - I've asked you a question about a possible missing test 😄

File: __tests__/app.test.js (hunk 2 above)
Comment (Mpanasetckiy) [reply __tests__/app.test.js:140]: it's a good question. Actually, I didn't check if the topic exists in db in the first place. I am sure db will yield an empty array anyway. So, should I just add a test for "paper" topic?

**Pull request merged**